src/brsTypes/components/ComponentFactory.ts

```diff
@@ -34,11 +35,33 @@ export enum BrsComponentName {
 
 // TODO: update with more components as they're implemented.
 export class ComponentFactory {
+    private static additionalComponents = new Map<string, (name: string) => RoSGNode>();
+
+    /**
+     * Adds additional components types to the factory, so other software can extend brs if necessary.
+     * This would allow other software using this to add other node/component types at runtime
+     * For example, adding custom implementations of the built-in types, or
+     * adding additional types (PinPad, BusySpinner, etc) that aren't here yet
+     *
+     * @static
+     * @param {[componentType: string, ctor: (name: string) => RoSGNode][]} types
+     * @memberof ComponentFactory
```

Comment:
Since this is a typescript file, `@memberof` and the `{[componentType: string, ctor: (name: stirng) => RoSGNode][]}` parts of these JSDoc lines aren't necessary — the same information shows up in `public static` and the signature of the function on line 50 😄 

```suggestion
     * @param types - the types to add
```